### PR TITLE
feat: add jwks support

### PR DESCRIPTION
**Pull request by hf**
Adds support for verifying end-user JWTs using JWKS in addition to the JWT secret. This means that third-party authentication systems that use asymmetric JWTs can be used with Realtime.

### CODE REVIEW

**Review by filipecabaco — APPROVED** [review]

**Review by philss — COMMENTED** [review]
It looks good to me!

I only miss a test covering the new JWKs support. I think it would be nice to have something more "integrated" - around the `RealtimeWeb.JwtVerification.verify/3` - to show how the verification works in practice. WDYT?

File: lib/realtime_web/channels/realtime_channel.ex [hunk 1]
```diff
@@ -438,12 +438,14 @@ defmodule RealtimeWeb.RealtimeChannel do
     assign(socket, :access_token, tenant_token)
   end
 
-  defp confirm_token(%{assigns: %{jwt_secret: jwt_secret, access_token: access_token} = assigns}) do
+  defp confirm_token(%{assigns: assigns}) do
+    %{jwt_secret: jwt_secret, access_token: access_token} = assigns
+    jwt_jwks = Map.get(assigns, :jwt_jwks)
```
[the remark this answers is not in the document]
Comment (hf) [reply lib/realtime_web/channels/realtime_channel.ex:443]: No JWKS is not guaranteed to be there. I'm also new to Elxiir.

File: lib/realtime_web/channels/auth/channels_authorization.ex [hunk 2]
```diff
@@ -4,20 +4,20 @@ defmodule RealtimeWeb.ChannelsAuthorization do
   """
   require Logger
 
-  def authorize(token, secret) when is_binary(token) do
+  def authorize(token, jwt_secret, jwt_jwks) when is_binary(token) do
```
[the remark this answers is not in the document]
Comment (hf) [reply lib/realtime_web/channels/auth/channels_authorization.ex:7]: I'd rather keep the methods as strongly typed as possible at this stage. Took quite a bit of refactors to make it work.

**Pull request merged**